fe/fe-core/src/main/java/com/starrocks/external/iceberg/ExpressionConverter.java

```diff
@@ -4,110 +4,227 @@
+     *   lessThan -> BinaryPredicate(BinaryPredicate.Operator.LT, ..., ...)
+     *   lessThanOrEqual -> BinaryPredicate(BinaryPredicate.Operator.LE, ..., ...)
+     *   greaterThan -> BinaryPredicate(BinaryPredicate.Operator.GT, ..., ...)
+     *   greaterThanOrEqual -> BinaryPredicate(BinaryPredicate.Operator.GE, ..., ...)
+     *   in -> InPredicate(..., ..., false)
+     *   notIn -> InPredicate(..., ..., true)
+     *   startsWith -> FunctionCallExpr("starts_with", ...) or LikePredicate(Operator.LIKE, ..., "prefix%")
+     *
+     * Supported expression operations are:
+     *   and -> CompoundPredicate(and, ..., ...)
+     *   or -> CompoundPredicate(or, ..., ...)
+     *   not -> CompoundPredicate(not, ...)
+     *
+     * Constant expressions are(Automatically merged in StarRocks, can be ignored):
+     *   alwaysTrue
+     *   alwaysFalse
+     */
+    public static Expression toIcebergExpression(Expr expr) {
+        if (expr == null) {
             return null;
         }
 
-        BinaryPredicate predicate = (BinaryPredicate) expr;
-        Operation op = getIcebergOperator(predicate.getOp());
-        if (op == null) {
+        if (expr instanceof CompoundPredicate) {
+            CompoundPredicate compoundPredicate = (CompoundPredicate) expr;
+            CompoundPredicate.Operator op = compoundPredicate.getOp();
+            if (op == CompoundPredicate.Operator.NOT) {
+                if (compoundPredicate.getChild(0) instanceof FunctionCallExpr ||
+                        compoundPredicate.getChild(0) instanceof LikePredicate) {
+                    // TODO: No negation for operation: STARTS_WITH in the Apache Iceberg 0.12.1
+                    return null;
+                }
+                Expression expression = toIcebergExpression(compoundPredicate.getChild(0));
+                if (expression != null) {
+                    return not(expression);
+                }
+            } else {
+                Expression left = toIcebergExpression(compoundPredicate.getChild(0));
+                Expression right = toIcebergExpression(compoundPredicate.getChild(1));
+                if (left != null && right != null) {
+                    return (op == CompoundPredicate.Operator.OR) ? or(left, right) : and(left, right);
+                }
+            }
             return null;
         }
 
-        if (!(predicate.getChild(0) instanceof SlotRef)) {
+        Expr columnExpr = expr.getChild(0);
+        if (columnExpr == null) {
             return null;
         }
-        SlotRef ref = (SlotRef) predicate.getChild(0);
 
-        if (!(predicate.getChild(1) instanceof LiteralExpr)) {
+        String columnName;
+        if (columnExpr instanceof SlotRef) {
+            columnName = ((SlotRef) columnExpr).getDesc().getColumn().getName();
+        } else if (columnExpr instanceof CastExpr) {
+            CastExpr columnCastExpr = (CastExpr) columnExpr;
+            if (!(columnCastExpr.getChild(0) instanceof SlotRef)) {
```

Comment:
What if multi-level nested cast? Like `cast(cast(xxx as varchar) as int)` ?

Reply:
This is just an optimization option, multi-level nesting is not considered at present, and correctness has a higher priority than covering more cases.

Reply:
Fine.